api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java

```diff
@@ -234,45 +234,55 @@
     long timeLimit() default 0;
 
     /**
-     * @return the unit of time that the {@link LRA#timeLimit()} attribute is
+     * @return the unit of time that the {@link LRA#timeLimit()} element is
      * measured in.
      */
     ChronoUnit timeUnit() default ChronoUnit.SECONDS;
 
     /**
      * <p>
-     * Normally if an LRA is present when a bean method is executed it will not
-     * be ended when the method returns. To override this behaviour and force LRA
-     * termination on exit use the end element.
+     * Normally if an LRA is present when a resource method is executed it will not
+     * be ended when the method returns. To override this behaviour and force the LRA
+     * to close on exit use this <code>end</code> element with the value set to
+     * {@literal true}.
      * </p>
      *
      * <p>
-     * If the <code>end</code> value is set to <code>false</code>
-     * while the annotated method returns the {@link Response} status code
-     * with the meaning the LRA to be cancelled (see {@link #cancelOn()} and {@link #cancelOnFamily()})
-     * then this error state has precedence over the <code>end</code> attribute.
-     * It means the LRA will be cancelled despite the end was set to <code>false</code>.
+     * If the <code>end</code> value is set to {@literal false} but the annotated
+     * method finishes with a status code that matches any of the values specified
+     * in the {@link #cancelOn()} or {@link #cancelOnFamily()} elements
+     * then the  LRA will be cancelled. In other words the
+     * {@link #cancelOn()} and {@link #cancelOnFamily()} elements take precedence
+     * over the <code>end</code> element.
      * </p>
      *
-     * @return true if an LRA that was present before method execution will be
-     * terminated when the bean method finishes.
+     * @return true if an LRA that was active when the method ran should be closed
+     * when the method execution finishes.
      */
     boolean end() default true;
 
     /**
      * <p>
      * The cancelOnFamily element can be set to indicate which families of
-     * HTTP response codes will cause the current LRA to cancel.
-     * This expects the annotated method is the JAX-RS endpoint where the {@link Response}
-     * with the status code could be returned.
+     * HTTP response codes will cause the current LRA to cancel. If the LRA
+     * has already been closed when the annotated method returns then this
```

Comment:
In which circumstances is it possible that the LRA is already closed?

Are there any other that the following one?

Method A calls Method B (with B having type=Required for example, not starting a new LRA)
Method B has end=true
So when Service A ends, LRA is already closed.

Reply:
Marking method B with any of MANDATORY, REQUIRED and SUPPORTS with end=true or if method B calls other methods similarly annotated then the LRA will be closed. The Java API would have been another mechanism for causing it to be closed.

The text I have written is a catch all and listing each possible scenario is risky since we may omit one. But if you want me to include examples then let me know.

Reply:
@mmusgrov Thanks for confirmation. I wasn't sure if there was any other possibility.  No need to add examples. Thx